[Docs] add github sync repo migration guide (#11805)
PipedreamHQ/pipedream: docs-v2/pages/projects/git.mdx

No, at this time it's not possible because of how resources are connected during the bootstrapping process from the workflow YAML specification.
However, this is on our roadmap, [please subscribe to this issue](https://github.com/PipedreamHQ/pipedream/issues/9255) for the latest details.

> ### Can I migrate my commit history to another GitHub Repository while retaining my workflows in my Pipedream Project?

Reviewer: Suggestion to simplify the title

```suggestion
### Migrating Github Repositories
```

> Yes, this guide shows how to migrate a GitHub-synced project repository in Pipedream to a new repository, preserving the entire commit history in the `production` branch while ensuring that workflows remain unaffected.

Reviewer: ```suggestion
You can migrate Pipedream project's Github repository to a new repository, while preserving history. You may want to do this when migrating a repository from a personal Github account to an organization account, without affecting the workflows within the Pipedream project.
```

#### Assumptions
- **Current GitHub Repository**: `previous_github_repo`
- **New GitHub Repository**: `new_github_repo`

> - **Pipedream Project**: `pipedream_project`

Reviewer: I don't think you'll need to specify the Pipedream project by name, there's only one project.

- Basic familiarity with git and GitHub
- Access to a local terminal (e.g., Bash, Zsh, PowerShell)
- Necessary permissions to modify both the Pipedream project and associated GitHub repositories

#### Steps

1. **Access Project Settings in Pipedream:**
   - Navigate to your `pipedream_project`.

>    - Use the dropdown menu on the "Edit" button in the top right corner to access `previous_github_repo` in GitHub.

Reviewer: Could use a screenshot

2. **Clone the Current Repository Locally:**
   ```
   git clone previous_github_repo_clone_url
   ```
3. Reset GitHub Sync in Pipedream:
   - In Pipedream, go to your project settings.

>    - Click on "Reset GitHub Connection".

Reviewer: Anytime there's a  click action, it's a good idea to include a screenshot for a visual reference

4. Set Up New Repository Connection:

>    - Configure a new GitHub Sync to `new_github_repo`.

Reviewer: ```suggestion
   - Configure a new GitHub Sync to `new_github_repo`.
```

- Configure the project's Github repository to use `new_github_repo`.

> 5. Clone the New Repository Locally:

Reviewer: Only headings need capitalization on each word, you can remove the captialization for words:

```suggestion
5. Clone the new repository locally:
```

   ```
   git clone new_github_repo_clone_url
   cd new_github_repo
   ```

> 6. Link to the Old Repository:

Reviewer: ```suggestion
6. Link to the old Repository:
```

#### Assumptions
- **Current GitHub Repository**: `previous_github_repo`
- **New GitHub Repository**: `new_github_repo`
- Basic familiarity with git and GitHub
- Access to a local terminal (e.g., Bash, Zsh, PowerShell)
- Necessary permissions to modify both the Pipedream project and associated GitHub repositories

#### Steps

1. **Access Project Settings in Pipedream:**
   - Navigate to your Pipedream project.
   - Use the dropdown menu on the "Edit" button in the top right corner to access `previous_github_repo` in GitHub.

   ![Accessing GitHub repository](https://res.cloudinary.com/pipedreamin/image/upload/v1714749205/Screenshot_2024-05-03_at_12.12.36_PM_sxkrng.png)

2. **Clone the Current Repository Locally:**
   ```
   git clone previous_github_repo_clone_url
   ```
3. Reset GitHub Sync in Pipedream:
   - In Pipedream, go to your project settings.
   - Click on "Reset GitHub Connection".

   ![Resetting GitHub connection](https://res.cloudinary.com/pipedreamin/image/upload/v1714749323/Screenshot_2024-05-03_at_12.15.03_PM_icdppb.png)

> 4. Set Up New Repository Connection:

Reviewer: All of these numbered steps should have normal capitalization:

 ```suggestion
4. Set Up New repository connection:
```

   - Configure the project's GitHub repository to use `new_github_repo`.
5. Clone the new repository locally:
   ```
   git clone new_github_repo_clone_url
   cd new_github_repo
   ```

> 6. Link to the old Repository:

Reviewer: Same thing:

```suggestion
6. Link to the old repository:
```

   ```
   git remote add old_github_repo previous_github_repo_clone_url
   git fetch --all
   ```

> 7. Prepare for Migration:

Reviewer: ```suggestion
7. Prepare for migration:
```

   - Create and switch to a new branch for migration:
      ```
      git checkout -b migration
      ```
   - Merge the main branch of `old_github_repo` into migration, allowing for unrelated histories:
      ```
      git merge --allow-unrelated-histories old_github_repo/production
      # Resolve any conflicts, such as in README.md
      git commit
      ```

> 8. Finalize Migration:

Reviewer: ```suggestion
8. Finalize the migration:
```
